### PR TITLE
feat: add treasury page

### PR DESCRIPTION
**Pull request by yathindrak**
Closes: https://github.com/DXgovernance/davi-monorepo/issues/41

### CODE REVIEW

**Comment by yathindrak** [comment]
Treasury page with empty token balances:
Screenshot 2023-02-01 at 19 28 21

Treasury page with token balances:
Screenshot 2023-02-01 at 18 14 08

**Comment by Kenny-Gin1** [comment]
Looks really solid @yathindrak! Good job! Will review the code further in a bit. Are there any permissions/restrictions that should be put in place when looking at a treasury. I noticed in the screenshots above that you didn't need to be part of the guild for example. Is that the intended behaviour @rossneilson?

File: apps/davi/src/Modules/Guilds/pages/Treasury/Treasury.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const { t } = useTranslation();
+  const { guildId: daoAddress } = useTypedParams();
+
+  const {
+    data,
+    isLoading: isBalancesLoading,
+    isError: isErrorRetrievingBalances,
+  } = useAllERC20Balances(daoAddress, true);
+
+  console.log({ isBalancesLoading });
```
Comment (dcrescimbeni) [inline apps/davi/src/Modules/Guilds/pages/Treasury/Treasury.tsx:70]:
```suggestion
```

**Review by dcrescimbeni — COMMENTED** [review]
Left just one comment about a console log. The rest looks good!

**Comment by dcrescimbeni** [comment]
I think we might need to refactor the loading part.

I get to see the loading indicator and the (empty) assets table before it loads:

image

**Comment by mprasanjith** [comment]
> Looks really solid @yathindrak! Good job! Will review the code further in a bit. Are there any permissions/restrictions that should be put in place when looking at a treasury. I noticed in the screenshots above that you didn't need to be part of the guild for example. Is that the intended behaviour @rossneilson?

@Kenny-Gin1 I think the treasury page should be public anyway. Its just stuff you can see publicly on-chain.

**Comment by yathindrak** [comment]
@dcrescimbeni Thanks for pointing out the issue. Done a fix in the latest commit.

**Review by dcrescimbeni — APPROVED** [review]
Looks good!

Just a thought for later: to get all the token balances, we're spamming the RPC through `useAllTokenBalances,` then filtering each token that has zero balance.

Probably in the future, we should fetch the treasury data from the subgraph.

**Pull request merged**